spacelift/resource_stack_destructor.go

```diff
@@ -0,0 +1,132 @@
+package spacelift
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shurcooL/graphql"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+)
+
+func resourceStackDestructor() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceStackDestructorCreate,
+		ReadContext:   resourceStackDestructorRead,
+		UpdateContext: resourceStackDestructorUpdate,
+		DeleteContext: resourceStackDestructorDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"stack_id": {
+				Type:        schema.TypeString,
+				Description: "ID of the stack to delete and destroy on destruction",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"deactivated": {
+				Type:        schema.TypeBool,
+				Description: "If set to true, destruction won't delete the stack",
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func resourceStackDestructorCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	d.SetId(d.Get("stack_id").(string))
+	d.Set("deactivated", d.Get("deactivated"))
+	d.Set("stack_id", d.Get("stack_id"))
+
+	return resourceStackDestructorRead(ctx, d, meta)
+}
+
+func resourceStackDestructorRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var query struct {
+		Stack *structs.Stack `graphql:"stack(id: $id)"`
+	}
+
+	variables := map[string]interface{}{"id": graphql.ID(d.Get("stack_id").(string))}
+
+	if err := meta.(*internal.Client).Query(ctx, &query, variables); err != nil {
+		return diag.Errorf("could not query for stack: %v", err)
+	}
+
+	stack := query.Stack
+	if stack == nil {
+		d.SetId("")
+		return nil
+	}
+
+	return nil
+}
+
+func resourceStackDestructorUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	d.Set("deactivated", d.Get("deactivated"))
+
+	return resourceStackRead(ctx, d, meta)
+}
+
+func resourceStackDestructorDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if deactivated := d.Get("deactivated"); deactivated != nil && deactivated.(bool) {
+		d.SetId("")
+		return nil
+	}
+
+	var mutation struct {
+		DeleteStack *structs.Stack `graphql:"stackDelete(id: $id, destroy: $destroy)"`
```

Comment:
I don't think you need to pass "destroy" as a variable here?

Reply:
Also, should it not be called `destroyResources`?

Reply:
I don't think I understand what you mean, and it's called destroy.

Reply:
Ah, I can pass the const directly, right

Reply:
fixed